cmd/gardenlet/app/migration.go

```diff
@@ -106,3 +118,23 @@ func migrateDeprecatedTopologyLabels(ctx context.Context, log logr.Logger, seedC
 
 	return flow.Parallel(taskFns...)(ctx)
 }
+
+// TODO(Wieneo): Remove this function when feature gate RemoveAPIServerProxyLegacyPort is removed
+func migrateRemoveAPIServerProxyLegacyPortFeatureGate(ctx context.Context, log logr.Logger, gardenClient client.Client, seedName string) error {
```

Comment:
Right now, this function only validates the feature gate enablement, but it doesn't "migrate" anything.
Would you mind renaming the function to something more meaningful?